boards/arm/stm32h747i_disco/stm32h747i_disco_m7.dts

```diff
@@ -19,6 +19,7 @@
 		zephyr,shell-uart = &usart1;
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
+		zephyr,flash-controller = &mt25ql512ab1;
```

Comment:
That's a board _chosen_ definition, for sure.
That could be a part of the board overlay but most of the other platforms have this in their board.dts file.

Reply:
I agree this is a grey area.
Boards porting guidelines states that boards default configuration should take advantage of any "fancy" hardware available in the BSP: "Since an external flash is available, let's take advantage for storage of it an keep internal flash for application binary."
That's debatable of course.